basyx.aasrepository/basyx.aasrepository-core/src/test/java/org/eclipse/digitaltwin/basyx/aasrepository/AasRepositorySuite.java

```diff
@@ -72,9 +73,91 @@ public void allAasRetrieval() throws Exception {
 		List<AssetAdministrationShell> expected = DummyAasFactory.createShells();
 		AasRepository aasRepo = getAasRepository(expected);
 		PaginationInfo pInfo = new PaginationInfo(2, null);
-		Collection<AssetAdministrationShell> coll = aasRepo.getAllAas(pInfo).getResult();
+		AasFilter aasFilter = new AasFilter();
+		Collection<AssetAdministrationShell> coll = aasRepo.getAllAas(pInfo, aasFilter).getResult();
 		assertEquals(expected, coll);
 	}
+	
+	@Test
+	public void retrieveAllAasWithNoFiltering() throws Exception {
+		List<AssetAdministrationShell> expected = DummyAasFactory.createShellsForFiltering();
+		AasRepository aasRepo = getAasRepository(expected);
+		PaginationInfo pInfo = PaginationInfo.NO_LIMIT;
+		AasFilter aasFilter = new AasFilter();
+		Collection<AssetAdministrationShell> coll = aasRepo.getAllAas(pInfo, aasFilter).getResult();
+		
+		assertTrue(expected.containsAll(coll));
+		assertTrue(coll.containsAll(expected));
+	}
+	
+	@Test
+	public void retrieveAllAasWithIdShortFiltering() throws Exception {
+		List<AssetAdministrationShell> allShells = DummyAasFactory.createShellsForFiltering();
+		AasRepository aasRepo = getAasRepository(allShells);
+		
+		List<AssetAdministrationShell> expectedShells = DummyAasFactory.createShellsWithSameIdShort();
+		
+		PaginationInfo pInfo = PaginationInfo.NO_LIMIT;
+		AasFilter aasFilter = new AasFilter();
+		aasFilter.setIdShort(DummyAasFactory.DUMMY_ID_SHORT);
+		
+		Collection<AssetAdministrationShell> coll = aasRepo.getAllAas(pInfo, aasFilter).getResult();
+		
+		assertTrue(expectedShells.containsAll(coll));
```

Comment:
This is not correctly asserting the filtering. Consider a different assertion approach - e.g. containsExactlyAll. There are several instances of this same issue in this file.